qutip/core/operators.py

```diff
@@ -632,9 +714,10 @@ def squeeze(N, z, offset=0):
      [ 0.00000000+0.j -0.30142443+0.j  0.00000000+0.j  0.95349007+0.j]]
 
     """
+    dtype = to.str2type(dtype or _data.Dense)
     asq = destroy(N, offset=offset) ** 2
     op = 0.5*np.conj(z)*asq - 0.5*z*asq.dag()
-    return op.expm()
+    return op.expm().to(dtype)
```

Comment:
In theory the right way to do this is to pass the out `dtype` to `expm` and let the data-layer do its thing in the normal manner, rather than having an explicit conversion step.